grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -1054,21 +1054,13 @@ class StockMovementService {
 
     List<ReceiptItem> getRequisitionBasedStockMovementReceiptItems(def stockMovement) {
         def shipments = Shipment.findAllByRequisition(stockMovement.requisition)
-        List<ReceiptItem> receiptItems = shipments*.receipts*.receiptItems?.flatten()?.sort { a, b ->
-            a.shipmentItem?.requisitionItem?.orderIndex <=> b.shipmentItem?.requisitionItem?.orderIndex ?:
-                    a.shipmentItem?.sortOrder <=> b.shipmentItem?.sortOrder ?:
-                            a?.sortOrder <=> b?.sortOrder
-        }
+        List<ReceiptItem> receiptItems = shipments*.receipts?.flatten()*.sortReceiptItemsBySortOrder()?.flatten()
```

Comment:
Is the flatten at the end required?

Reply:
I think the one in the middle is unnecessary, but the one on the end is necessary.  I'm starting to think this just needs to be a query in a data service getSortedReceiptItems(). But that can be a refactoring for another day.

Reply:
both are necessary.
Notice that for `getShipmentBasedStockMovementReceiptItems()` it has `shipment` and for `getRequisitionBasedStockMovementReceiptItems()` it is `shipments`